internal/balancer/balancer.go

```diff
@@ -175,6 +175,34 @@ func (b *Balancer) clusterDiscoveryAttempt(ctx context.Context, cc *grpc.ClientC
 	return nil
 }
 
+func buildConnectionsState(ctx context.Context, pool interface {
+	GetIfPresent(endpoint endpoint.Endpoint) conn.Conn
+	Allow(ctx context.Context, cc conn.Conn)
+	EndpointsToConnections(endpoints []endpoint.Endpoint) []conn.Conn
+}, newest []endpoint.Endpoint,
+	dropped []endpoint.Endpoint,
+	config balancerConfig.Config,
+	selfLocation balancerConfig.Info,
+) *connectionsState {
+	connections := pool.EndpointsToConnections(newest)
```

Comment:
Do I understand correctly that it will create a real connection for all endpoints here?

If so, it might be a problem for large clusters.

Reply:
added a new trace.Driver hook, connection is not pinged on discovery now